tests/client_request.rs

```diff
@@ -630,6 +630,34 @@ fn recv_too_big_headers() {
 
 }
 
+#[test]
+fn request_without_path() {
+    let _ = ::env_logger::try_init();
+    let (io, srv) = mock::new();
+
+    let srv = srv.assert_client_handshake()
+        .unwrap()
+        .recv_settings()
+        .recv_frame(frames::headers(1).request("GET", "http://example.com/").eos())
+        .send_frame(frames::headers(1).response(200).eos())
+        .close();
+
+    let client = client::handshake(io)
+        .expect("handshake")
+        .and_then(move |(mut client, conn)| {
+            // Note the lack of trailing slash.
+            let request = Request::get("http://example.com")
```

Comment:
It would be good to see tests for "OPTIONS *" and "GET *" as well as "OPTIONS /*" and "GET /*"

Reply:
I tried to write a test for OPTIONS, but there seems to be some problems unrelated to this change. So, I will open a new issue to track this.

Reply:
Ok, I figured out how to write a test, but #229 still applies as it is kind of janky.